database.cc

```diff
@@ -202,16 +202,20 @@ database::database(const db::config& cfg, database_config dbcfg)
         max_inactive_queue_length(),
         [this] {
             ++_stats->sstable_read_queue_overloaded;
-            return std::make_exception_ptr(std::runtime_error("sstable read queue overloaded"));
+            return std::make_exception_ptr(std::runtime_error("_read_concurrency_sem overloaded"));
         })
     // No timeouts or queue length limits - a failure here can kill an entire repair.
     // Trust the caller to limit concurrency.
     , _streaming_concurrency_sem(
             max_count_streaming_concurrent_reads,
-            max_memory_streaming_concurrent_reads())
+            max_memory_streaming_concurrent_reads(),
+            std::numeric_limits<size_t>::max(),
+            [] { return std::make_exception_ptr(std::runtime_error("_streaming_concurrency_sem overloaded")); })
     , _system_read_concurrency_sem(
             max_count_system_concurrent_reads,
-            max_memory_system_concurrent_reads())
```

Comment:
#5281 is also about changing timeout errors, so you need to propagate the name into `reader_concurrency_semaphore` and into `expiring_fifo`.

Also, there is more semaphores which deserve naming, e.g. the ones in the commitlog and region_group::run_when_memory_available()

Reply:
Part 1: Absolutely my bad, I didn't notice that it's timed. I'll do it.
Part 2: Sure, I can do it, but I'll need the task to indicate which semaphores are expected to be named. I don't want this PR to hang forever with debates on name for every mutex...